src/GoogleAdWordsEventForwarder.js

```diff
@@ -200,27 +295,58 @@
             return mParticle.generateHash(preHash);
         }
 
+        function loadGtagSnippet() {
+            (function () {
+                window.dataLayer = window.dataLayer || [];
+                window.gtag = function(){dataLayer.push(arguments);}
+
+                var gTagScript = document.createElement('script');
+                gTagScript.async = true;
+                gTagScript.onload = function () {
+                    gtag('js', new Date());
+                    gtag('config', gtagSiteId);
+                };
+                gTagScript.src = 'https://www.googletagmanager.com/gtag/js?id=' + gtagSiteId;
+                document.getElementsByTagName('head')[0].appendChild(gTagScript);
+            })();
+        }
+
+        function loadLegacySnippet() {
+            (function () {
+                var googleAdwords = document.createElement('script');
+                googleAdwords.type = 'text/javascript';
+                googleAdwords.async = true;
+                googleAdwords.onload = function() {
+                    if (eventQueue.length) {
```

Comment:
This flush logic could be factored out into a shared function for legacy and gtag